Feature/addsMarketsandStoresFilterButtons

cordhernandez opened the pull request:
+ adjusts the buttons border width
+ adjusts the buttons corner radius
+ adjusts the buttons background color
+ calls styleMenu() in viewDidLoad()
+ the logic/delegate code for these buttons will be added in another ticket
#143

cordhernandez committed:
adjusts UIButton colors in Filter

cordhernandez committed:
adds UIButton Outlets and Style Menu Code

cordhernandez committed:
creates IBActions for Filter Buttons

cordhernandez committed:
implements IBActions Code

cordhernandez committed:
minor UI updates

adjusts Filter Button colors

cordhernandez committed:
adjusts the filter buttons

+ adjusts the buttons border width
+ adjusts the buttons corner radius
+ adjusts the buttons background color
+ calls styleMenu() in viewDidLoad()

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -216,16 +263,30 @@ extension FilterViewController {
-        let center = mapView.centerCoordinate
+        farmersMarketsButton.layer.borderColor = UIColor.white.cgColor
+        farmersMarketsButton.layer.borderWidth = 1
+        farmersMarketsButton.layer.cornerRadius = 7
```
This is the sort of thing that can be done from Storyboard by using `@IBDesignable`.
It is recommended to do visual stuff in the visual editor, to maximize productivity and minimize code footprint.

Also, the cornerRadius is supposed to be 5. :wink:

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -216,16 +263,30 @@ extension FilterViewController {
+    
+    func styleButtonOn(button: UIButton) {
+        
+        button.layer.backgroundColor = UIColor.init(red: 0.902, green: 0.73, blue: 0.25, alpha: 1).cgColor
```
These color values seem guessed at. You definitely want to use code that lets you put in colors using HEX value.

Check out https://github.com/RedRoma/RedRomaColors:

https://github.com/RedRoma/RedRomaColors/blob/develop/RedRomaColors/RedRomaColors/Colors.swift

`Colors.fromRGB()`
`Colors.from(hexString:"")`

cordhernandez committed:
adds Colors class

adds fromRGB() and fromRGBA() methods

cordhernandez committed:
Removes old code and adds fromRGB() method

cordhernandez committed:
changes group folder name to Colors

cordhernandez committed:
cleans code

adds space

cordhernandez committed:
adds Fonts programmatically and IBDesignable

cordhernandez committed:
adds Custom Fonts and Button Animations

cordhernandez committed:
Modifies RoundedButtonView.swift

adds layer.borderWidth = borderWidth
adds layer.borderColor = borderColor.cgColor

cordhernandez commented:
need an admin to check this branch. thanks.

SirWellington commented on BlackNectar/BlackNectar/Colors.swift:
```diff
@@ -0,0 +1,26 @@
+
+public class Colors {
+    
+    public static func fromRGB(red: CGFloat, green: CGFloat, blue: CGFloat) -> UIColor {
```
Nice!

cordhernandez replied:
👊🏾

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -17,12 +17,14 @@ import UIKit
     var distance = 0.0
-    var showFarmersMarkets = true
-    var showGroceryStores = true
+    var showFarmersMarkets = false
```
The best option would be to load whatever the user has to set to default from `UserDefaults`

cordhernandez replied:
will do this in ticket #149

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -216,16 +262,21 @@ extension FilterViewController {
-        LOG.debug("User dragged Map Screen to: \(center)")
+        button.backgroundColor = Colors.fromRGB(red: 235, green: 191, blue: 77)
+        button.titleLabel?.font = UIFont(name: "Oxygen-Bold", size: 16)
+        UIView.transition(with: button, duration: 0.4, options: .transitionCrossDissolve, animations: nil, completion: nil)
```
Technically....the code that changes the buttons should be in an animations closure. That closure would then be passed to the UIView.transition method.

```swift
let animations = {
    button.backgroundColor = Colors.fromRGB(red: 235, green: 191, blue: 77)
    button.titleLabel?.font = UIFont(name: "Oxygen-Bold", size: 16)
}
UIView.transition(with: button, duration: 0.4, options: .transitionCrossDissolve, animations: animations, completion: nil)
```

cordhernandez replied:
Ohh ok this makes sense 
better to put animations instead of nil lol

cordhernandez replied:
job done

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -216,16 +262,21 @@ extension FilterViewController {
         
-        self.loadStoresInMapView(at: center)
+        button.backgroundColor = UIColor.clear
+        button.titleLabel?.font = UIFont(name: "Oxygen-Regular", size: 16)
```
These fonts should ideally be in their own class `Fonts`, so  you can quickly load them as variables, and not have to enter the String name here all the time.

cordhernandez replied:
alright now it looks clean 
job done

SirWellington reviewed (approved):
:green_apple:

cordhernandez committed:
adjusts groups and files

changes Fonts to Resources
adds new Fonts group

cordhernandez committed:
adds animation closures for buttons

cordhernandez committed:
adds Fonts.swift

adds oxygenBold
adds oxygenRegular

cordhernandez commented:
changes have been made 
please check 
thanks

SirWellington commented on BlackNectar/BlackNectar/Fonts.swift:
```diff
@@ -0,0 +1,20 @@
+    static let instance = Fonts()
+    private init() {}
+    
+    var oxygenBold = UIFont(name: "Oxygen-Bold", size: 16)
```
Wonderful!
You could probably skip the whole singleton pattern here and just have these be static constant variables.
`Fonts.oxygenBold`, etc

cordhernandez replied:
ahh that's what it was 
cool this is done

SirWellington commented on BlackNectar/BlackNectar/FilterViewController.swift:
```diff
@@ -216,16 +262,29 @@ extension FilterViewController {
+        
+        let animations = {
+            button.backgroundColor = UIColor.clear
+            button.titleLabel?.font = Fonts.instance.oxygenRegular
```
This is much betta.

cordhernandez replied:
👍🏽

SirWellington reviewed (commented):
One minor tweak and we good

cordhernandez committed:
removes singleton and changes variables to static constants

cordhernandez committed:
modifies Fonts

removes singleton
